### PR TITLE
proot: update to 5.2.0.

### PR DESCRIPTION
**Pull request by Goorzhel**
- Delete obsolete patches and fix the rest.
- Add pkg-config (otherwise, undefined talloc_* refs abound).
- Use docs/GNUMakefile to generate manpage.

I could still use some help getting this to cross-compile on ARM architectures; whenever I try it, I get `pkg-config: No such file or directory`. It could be the case that I'm simply using `pkg-config` wrong (my C isn't great, to be honest).

#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [X] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by Goorzhel** [comment]
Another foible: `src/Makefile` depends on CWD being a Git repo:
```
GIT_VERSION := $(shell git describe --tags `git rev-list --tags --max-count=1`)

GIT_COMMIT := $(shell git rev-list --all --max-count=1 | cut -c 1-8)

VERSION = $(GIT_VERSION)-$(GIT_COMMIT)
```
So the Void package, being built from a tarball, does this —
```
❯ proot --version | head -4
 _____ _____              ___
|  __ \  __ \_____  _____|   |_
|   __/     /  _  \/  _  \    _|
|__|  |__|__\_____/\_____/\____| -
```
— instead of `5.2.0-d0a02a66`.

**Comment by Goorzhel** [comment]
Rebasing for [new mirror](https://voidlinux.org/news/2021/10/mirror-retirement.html).

File: srcpkgs/proot/template [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-version=5.1.0
-revision=7
+version=5.2.0
+revision=1
 archs="x86_64* i686* aarch64* arm*"
 build_wrksrc=src
 build_style=gnu-makefile
 make_use_env=yes
-makedepends="libarchive-devel talloc-devel"
+makedepends="libarchive-devel talloc-devel python3-docutils pkg-config python3-lxml"
```
Comment (Piraty) [inline srcpkgs/proot/template:9]: those are host dependencies

**Review by Piraty — CHANGES_REQUESTED** [review]
please fix cross compilation

last time i heard about it, we discussed removing this package from the tree...

**Comment by Goorzhel** [comment]
No, it had worked when I compiled it in x64 but I didn't try it in ARM. 🤦‍♂️ _Now_ it should be good.

File: srcpkgs/proot/patches/disable-seccomp-based-tracing.patch [hunk 2]
```diff
@@ -1,28 +1,11 @@
...
-From: Simon McVittie <smcv@debian.org>
-Date: Sat, 14 Jan 2017 15:52:13 +0000
-X-Dgit-Generated: 5.1.0-1.2 dcc5f0999759be03aed5b62a8683e0b965d0219d
-Subject: Disable seccomp-based tracing performance improvement
-
-It is faster (according to upstream documentation), but on current
-(4.8.4+) kernels it just segfaults. Software that works slowly seems
-better than software that doesn't work at all.
-
-Bug: https://github.com/proot-me/PRoot/issues/106
```
Comment (Piraty) [inline srcpkgs/proot/patches/disable-seccomp-based-tracing.patch:13]: do not remove reference to issue

File: srcpkgs/proot/patches/disable-seccomp-based-tracing.patch [hunk 3]
```diff
@@ -1,28 +1,11 @@
-# reason: https://github.com/proot-me/proot/issues/106
```
Comment (Piraty) [inline srcpkgs/proot/patches/disable-seccomp-based-tracing.patch:1]: do not remove reference to origin

File: srcpkgs/proot/template [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
-version=5.1.0
-revision=7
+version=5.2.0
+revision=1
 archs="x86_64* i686* aarch64* arm*"
 build_wrksrc=src
 build_style=gnu-makefile
 make_use_env=yes
+make_build_args="VERSION=${version}"
+hostmakedepends="python3-docutils pkg-config python3-lxml"
```
Comment (Piraty) [inline srcpkgs/proot/template:10]: sort the entries

**Review by Piraty — CHANGES_REQUESTED** [review]
good to merge otherwise

**Pull request closed without merging**